packages/plugin-credentials/src/credentials/common.js

```diff
@@ -151,20 +173,19 @@ export default {
     }
 
     if (!scope) {
-      scope = apiScope;
+      //scope = apiScope;
+      scope = filterScope(`spark:kms`, this.config.scope);
     }
-
     scope = sortScope(scope);
-
     const token = this.userTokens.get(scope);
 
     if (!token) {
       return this.supertoken.downscope(scope)
+        .then((downscopeToken) => downscopeToken.toString())
         .catch((reason) => this._handleDownscopeFailure(this.supertoken, scope, reason))
         .then(tap((t) => this.userTokens.add(t)));
```

Comment:
I think the tap line needs to happen before the downscope line; doesn't usertokens.add here add the string rather than the token instance?